File: app/my_post_request.rb
class MyPostRequest < Com::Android::Volley::Request #::JsonObjectRequest

  def getHeaders
    headers = {}
    auth_string = @username + ":" + @password
    auth = "Basic " + convertTo64(auth_string)
    auth = Java::Lang::String.new(auth)
    headers[Java::Lang::String.new("Authorization")] = auth
    headers
  end

  def username=(username)
    @username = username
  end

  def password=(password)
    @password = password
  end

  def parseNetworkResponse(response)
    p "parseNetworkResponse"
    jsonString = response.data
    p jsonString.toString
    return Com::Android::Volley::Response.success(jsonString, Com::Android::Volley::Toolbox::HttpHeaderParser.parseCacheHeaders(response))
  end

  # def getParams
  #   params = {}
  #   params["barcode"] = "barcode"
  #   # params[Java::Lang::String.new("grant_type")] = Java::Lang::String.new("client_credentials")
  #   # params
  # end

  def deliverResponse(response)
    # p response
    p "deliverResponse"
    listener = getListener
    listener.onResponse(response)
  end

end

File: app/my_post_request.java
private java.util.Map<java.lang.String, java.lang.String> params;
private com.android.volley.Response.Listener<org.json.JSONObject> listener;

public MyPostRequest(int method, java.lang.String url, java.util.Map<java.lang.String, java.lang.String> params,
        com.android.volley.Response.Listener<org.json.JSONObject> reponseListener, com.android.volley.Response.ErrorListener errorListener) {
    super(method, url, errorListener);
    this.listener = reponseListener;
    this.params = params;
}

protected java.util.Map<java.lang.String, java.lang.String> getParams()
        throws com.android.volley.AuthFailureError {
    return params;
};

public com.android.volley.Response.Listener<org.json.JSONObject> getListener() {
  return this.listener;
}

public java.lang.String convertTo64(java.lang.String string) {
  return android.util.Base64.encodeToString(string.getBytes(),  android.util.Base64.NO_WRAP);
}

// @Override
// protected void deliverResponse(org.json.JSONObject response) {
//     // TODO Auto-generated method stub
//     listener.onResponse(response);
// };

// @Override
// protected void deliverResponse(java.lang.Object response) {
//     // TODO Auto-generated method stub
//     // listener.onResponse(response);
// };
